File: src/main/java/net/volcano/jdacommands/model/command/arguments/interfaces/CodecRegistry.java
package net.volcano.jdacommands.model.command.arguments.interfaces;

import javax.annotation.Nullable;

public interface CodecRegistry {
	
	@Nullable
	<T> Codec<T> getCodec(Class<T> clazz);
	
	<T> void registerCodec(Codec<T> codec);
	
}

File: src/main/java/net/volcano/jdacommands/model/command/arguments/interfaces/CodecRegistry.kt
package net.volcano.jdacommands.model.command.arguments.interfaces

import kotlin.reflect.KClass

interface CodecRegistry {

	fun <T : Any> getCodec(clazz: KClass<T>): Codec<T>?
	fun <T> registerCodec(codec: Codec<T>)

}